client/dashboard/components/formatted-time/index.ts

```diff
@@ -0,0 +1,27 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { useLocale } from '../../app/locale';
+import { formatDate } from '../../utils/datetime';
+
+export function useFormattedTime( timestamp: string, formatOptions?: Intl.DateTimeFormatOptions ) {
+	const locale = useLocale();
+
+	const date = new Date( timestamp );
+	const formatted = formatDate( date, locale, formatOptions );
```

Comment:
[nit] We can call it at the end of the hook 🙂

Reply:
Cannot because it's used inside `if ( isToday )`